src/components/RangeInput.vue

```diff
@@ -0,0 +1,268 @@
+<template>
+  <div :class="bem()">
+    <slot name="header" />
+
+    <form @submit.prevent="onSubmit(refinement)">
+      <input
+        type="number"
+        :class="bem('input', 'from')"
+        :min="range.min"
+        :max="range.max"
+        :step="step"
+        :placeholder="rangeForRendering.min"
+        :value="refinementForRendering.min"
+        @input="refinement.min = $event.target.value"
+      />
+
+      <slot name="separator">
+        to
+      </slot>
+
+      <input
+        type="number"
+        :class="bem('input', 'to')"
+        :min="range.min"
+        :max="range.max"
+        :step="step"
+        :placeholder="rangeForRendering.max"
+        :value="refinementForRendering.max"
+        @input="refinement.max = $event.target.value"
+      />
+      <slot name="submit">
+        <button>Ok</button>
+      </slot>
+    </form>
+
+    <slot name="footer" />
+  </div>
+</template>
+
+<script>
+import algoliaComponent from '../component';
+import { FACET_OR } from '../store';
+
+export default {
+  mixins: [algoliaComponent],
+  props: {
+    attributeName: {
+      type: String,
+      required: true,
+    },
+    min: {
+      type: Number,
+    },
+    max: {
+      type: Number,
+    },
+    defaultRefinement: {
+      type: Object,
+      default() {
+        return {};
+      },
+    },
+    precision: {
+      type: Number,
+      default: 0,
+      validator(value) {
+        return value >= 0;
+      },
+    },
+  },
+
+  data() {
+    return {
+      blockClassName: 'ais-range-input',
+    };
+  },
+
+  created() {
+    const { min: minValue, max: maxValue } = this.defaultRefinement;
+
+    let min;
+    if (minValue !== undefined) {
+      min = minValue;
+    } else if (this.min !== undefined) {
+      min = this.min;
+    }
+
+    let max;
+    if (maxValue !== undefined) {
+      max = maxValue;
+    } else if (this.max !== undefined) {
+      max = this.max;
+    }
+
+    this.searchStore.stop();
+
+    this.searchStore.addFacet(this.attributeName, FACET_OR);
+
+    if (min !== undefined) {
+      this.searchStore.addNumericRefinement(this.attributeName, '>=', min);
+    }
+
+    if (max !== undefined) {
+      this.searchStore.addNumericRefinement(this.attributeName, '<=', max);
+    }
+
+    this.searchStore.start();
+    this.searchStore.refresh();
+  },
+
+  destroyed() {
+    this.searchStore.removeFacet(this.attributeName);
+  },
+
+  computed: {
+    step({ precision }) {
```

Comment:
First time I see computed getters with an argument ;)
Cleans up those `this` everywhere, pretty cool.
Does Vue pass the vm down everywhere?

Reply:
On `computed` property the `vm` is always pass as first argument in addition to the context. I don't know if it's recommended or not to use the arguments vs `this`, Vue is pretty new to me.

But I found that a lot easier to test, no need to instantiate the the component.

Reply:
> On computed property the vm is always pass as first argument in addition to the context.

I was unable to find a single example using this syntax in the different projects I've seen.
Even in all official tests of Vue itself, `this` is used.
Given that the documentation and the API docs never speak of the fact that the vm is injected as first argument, I would tend to recommend sticking to using `this`. 
Vue can decide tomorrow to remove those parameters, because there not documented, they could say it's an internal concern.

Reply:
Using the vm as an argument was introduced quite recently but it's ok to use 🙂 
It's meant to be used with  arrow functions:

```js
step: ({ precision }) => 1 / Math.pow(10, precision)
```

It's not documented but it should